src/pages/SingleVideo/SingleVideo.jsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { SideNav } from "../../components";
+import ReactPlayer from "react-player/youtube";
+import "./SingleVideo.css";
+import {
+  getVideo,
+  likeVideo,
+  unlikeVideo,
+  addToWatchLater,
+  removeFromWatchLater,
+} from "../../utils";
+import { useAuth, useLikes, useWatchLater } from "../../context";
+
+export const SingleVideo = () => {
+  const [video, setVideo] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [liked, setLiked] = useState(false);
+  const [watchLater, setWatchLater] = useState(false);
+
+  const {
+    authState: { token },
+  } = useAuth();
+  const {
+    likesState: { likes },
+    likesDispatch,
+  } = useLikes();
+  const {
+    watchLaterState: { watchlater },
+    watchLaterDispatch,
+  } = useWatchLater();
+  const navigate = useNavigate();
+  const params = useParams();
+
+  const likeHandler = (_id) => {
+    if (token) {
+      likeVideo(token, video, likesDispatch, setLiked);
+    } else {
+      navigate("/login");
+    }
+  };
+
+  const unlikeHandler = (_id) => {
+    unlikeVideo(_id, token, likesDispatch, setLiked);
+  };
+
+  const addToWatchLaterHandler = (_id) => {
+    if (token) {
+      addToWatchLater(token, video, watchLaterDispatch, setWatchLater);
+    } else {
+      navigate("/login");
+    }
+  };
+
+  const removeFromWatchLaterHandler = (_id) => {
+    removeFromWatchLater(_id, token, watchLaterDispatch, setWatchLater);
+  };
```

Comment:
You could have kept these methods inside contexts and made it available in here as part of context value. This way you are keeping the context and functions that manipulate the context together.

```JSX
<VideoContext.Provider value={{addToLike, addToWatchLater}} >
    {children}
</VideoContext.Provider>
```